merkletree/proof.go

```diff
@@ -16,6 +23,71 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
```

Comment:
These proof changes are kind separate and, again, deserve a pull of their own.

Let's try to keep the habit of,

1. Making changes as small as possible;
2. Keeping logically separate changes to their own commits (which kind of follow from 1).

Having lots of small commits that we can merge independently is preferable.  It eases the cognitive load on the reviewer.